notebooks/Notebook1.md

```diff
@@ -18,17 +24,35 @@ exciting milestones, preliminary results, etc. Feel free to include images
 code, and any other resources that you think will help clearly convey your
 design process.
 
+This week has been a very interesting, design-wise. For the flexibility criterion I discussed above, I decided very quickly that my project should be an internal DSL. That way, in case there is some way in which the language is incomplete, a seasoned programmer could add some host language code as a stopgap until the DSL can be updated. This realization was difficult- I spent a few hours programming an object-oriented framework for board games, then extended it to 52-card-deck card games, then used that framework to program the game Hearts. The process of programming exposed a lot of the "sticking points" where a DSL could be useful. For example, many games have a set round structure, in which each player takes a turn; each of these turns is a specific sequence of steps, each of which are made of one or more actions. These actions are in turn composed of atomic moves, like moving a card from one location to another, or rolling dice. The below document outlines the current OOP structure of game objects, but but the Round-Turn-Action-Move relationship is currently eluding me.
+
```

Comment:
I don't yet fully understand what you're describing here, but let me know if you want to chat about it and brainstorm some implementation ideas!